spikeinterface/core/waveform_extractor.py

```diff
@@ -272,10 +280,101 @@ def get_waveforms_segment(self, segment_index, unit_id, sparsity=None):
             The returned waveform (num_spikes, num_samples, num_channels)
         """
         wfs, index_ar = self.get_waveforms(unit_id, with_index=True, sparsity=sparsity)
-        segment_index_ar = np.array([i[1] for i in index_ar])
-        return wfs[segment_index_ar == segment_index, :, :]
+        mask = index_ar['segment_index'] == segment_index
+        return wfs[mask, :, :]
+    
+    def precompute_templates(self, modes=('average', ), quantile_value=None):
+        """
+        Precompute all template for different "modes":
+          * average
+          * std
+          * median
+          * quantile
```

Comment:
@samuelgarcia I would remove the quantiles

@Saksham20 quantiles are not really used and makes it much more complicated to handle internally. If needed, one can always grab the waveforms and do whatever :)